spoon-runner/src/test/java/com/squareup/spoon/SpoonCoverageMergerTest.java

```diff
@@ -0,0 +1,58 @@
+package com.squareup.spoon;
+
+import org.hamcrest.CustomTypeSafeMatcher;
+import org.jacoco.core.tools.ExecFileLoader;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.io.File;
+import java.util.HashSet;
+
+import static java.util.Arrays.asList;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Matchers.argThat;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class SpoonCoverageMergerTest {
+
+  @Test
+  public void shouldMergeCoverageFiles() throws Exception {
+    ExecFileLoader execFileLoader = mock(ExecFileLoader.class);
+    HashSet<String> serials = new HashSet<String>(asList("10.0.0.1:1234", "10.0.0.2:1234"));
+    File spoonOutputDirectory = new File("/output");
+
+    SpoonCoverageMerger spoonCoverageMerger = new SpoonCoverageMerger(execFileLoader);
+
+    spoonCoverageMerger.mergeCoverageFiles(serials, spoonOutputDirectory);
+
+    ArgumentCaptor<File> captor = ArgumentCaptor.forClass(File.class);
+    verify(execFileLoader, times(2)).load(captor.capture());
+    assertThat(captor.getAllValues().get(0), coverageFileOf("10.0.0.1:1234"));
+    assertThat(captor.getAllValues().get(1), coverageFileOf("10.0.0.2:1234"));
+
+    verify(execFileLoader, times(1)).save(argThat(hasPath("/output/coverage/merged-coverage.ec")), eq(false));
```

Comment:
> Ideally, a unit test should never touch the file system otherwise this test might start failing if write permissions are not assigned to store file on the disk

This is a non-issue with the `TemporaryFile` JUnit rule.

I'm much more concerned about the fact that the test is white-box and will need changed with any change to `SpoonCoverageMergerTest` which is prone to breakage. Currently the unit test is an encoding of _how_ rather than _what_. Unit tests should verify behavior correctness and not implementation.